walkthroughs/howto-http-retries/README.md

```diff
@@ -0,0 +1,64 @@
+# HTTP Retry Policy example
+
+1. Clone this repository and navigate to the walkthrough/http-retry-policy folder, all commands will be ran from this location
+
+
+## Setup
+
+1. Your account id:
+    ```
+    export AWS_ACCOUNT_ID=<your_account_id>
+    ```
+
+2. Your region
+    ```
+    export AWS_DEFAULT_REGION=us-west-2
+    
+    ```
+3. The latest envoy image, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+   ```
+   export ENVOY_IMAGE=<from_appmesh_user_guide_above>
+   ```
+    
+4. Deploy the resources (this will take about 5-10 minutes to complete):
+    ```
+    ./deploy.sh
+    ```
+   
+5. Once the script has executed there will be a public endpoint, save this for later.
+We will use this endpoint in curl to pass header values to the header "statuscode_header" to observe retry policy in action.
+
+## Update Route
+
+1. Check the current route configuration calling describe-route by name. For instance, to describe the blue route (notice the retry policy in the spec):
+    ```
+    aws appmesh describe-route --mesh-name howto-http-retries --virtual-router-name color-router --route-name color-route-blue
+    ```
+    
+2. Use curl and send requests to your frontend service while adding headers to see retry policy in place. In the cloudformation template we set the
+retry limit to 4 so you should see 5 total failures in your logs. One for the initial request and then 4 retries.  
+    ```
+    curl -H "statuscode-header: 503" <endpoint> -v    
```

Comment:
Do you think we should have a step that first makes the request without the status code first? Then it's obvious that we're getting a 503, and then introduction of this retry solves that.

Reply:
In fact, just straight curling the endpoint without the header still succeeds. I don't think that's the right behavior is it?

Reply:
Good idea ive rewritten this section. That is the intended result, the route is valid so without the header we return 200 as defined in the color app service

Reply:
We chatted offline, but ideally the behavior is apparent when curling the endpoint. A potential idea is having the backend fail pretty often, and the demo defaults to no retries; introducing the retries would increase the success chance.